### PR TITLE
#549 Image Display element plugin

### PR DESCRIPTION
**Pull request by CarlosNZ**
Implements #549. A fairly basic version of an Image Display plugin.

_Branched from #555 -- please review that one first._

---

Rather than get too carried away with sizing styling, I've just restricted the "size" parameter to the same as Semantic-UI image "size" options:, i.e. `mini`, `tiny`, `small`, `medium`, `large`, `big`, `huge`, `massive`

### To Test:
- Use branch from [back-end PR #291](https://github.com/openmsupply/application-manager-server/pull/291).
- Fire up a **Feature Showcase** application and navigate to **Section2**, **Page 3**.
- Upload an image file in the File Upload element, and an ImageDisplay element will become visible, displaying this image. 
- Dynamic Size and Alignment controls for the Image Display are also available as Dropdown selectors below the image

**Note**: there is a hard-coded host value ("http://localhost:8080") in one of the image template expressions. We'll be able to make this dynamic once we implement "ApplicationData" ([#458](https://github.com/openmsupply/application-manager-web-app/issues/458)) in front-end.

### CODE REVIEW

File: documentation/internal/Element-Type-Specs.md [hunk 1]
```diff
@@ -157,7 +177,7 @@ _Multi-choice question, with one allowed option, displayed as Drop-down list (Co
...
 - **default**: `string`/`number` -- if not provided, defaults to index 0.
 - **search**: `boolean` (default: `false`) -- if `true`, the list of options can be searched and filtered by user
-- **optionsDisplayProperty**: If `options` (above) consists of an array of objects, this parameter specifies the field of each object to be displayed in the options list. For example, if `options` was a list of organisation objects (i.e. `{orgId, name, licenceNumber}`), you'd probably specify `name` as the `optionsDisplayProperty`. Note that even though one field is displayed to the user in the Dropdown list, the _entire_ selected object is saved as the selection.
+- **optionsDisplayProperty**: If `options` (above) consists of an array of objects, this parameter specifies the field of each object to be displayed in the options list. For example, if `options` was a list of organisation objects (i.e. `{orgId, name, licenceNumber}`), you`d probably specify `name`as the`optionsDisplayProperty`. Note that even though one field is displayed to the user in the Dropdown list, the _entire_ selected object is saved as the selection.
```
Comment (nmadruga) [inline documentation/internal/Element-Type-Specs.md:180]: Unintended?

File: src/formElementPlugins/pluginProvider.tsx [hunk 2]
```diff
@@ -45,6 +45,13 @@ class pluginProvider {
         folderName: 'textInfo',
         category: 'Informative',
       },
+      imageDisplay: {
+        code: 'imageDisplay',
+        isCore: true,
+        displayName: 'Image Display',
```
Comment (nmadruga) [inline src/formElementPlugins/pluginProvider.tsx:51]: Could be called just "Image"

File: src/formElementPlugins/imageDisplay/pluginConfig.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "isCore": true,
+  "code": "imageDisplay",
+  "displayName": "Image Display",
+  "dynamicParameters": ["label", "description", "url", "size", "alignment"],
```
Comment (nmadruga) [inline src/formElementPlugins/imageDisplay/pluginConfig.json:5]:
```suggestion
  "dynamicParameters": ["label", "description", "url", "size", "alignment", "altText"],
```

File: src/formElementPlugins/imageDisplay/pluginConfig.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "isCore": true,
+  "code": "imageDisplay",
+  "displayName": "Image Display",
```
Comment (nmadruga) [inline src/formElementPlugins/imageDisplay/pluginConfig.json:4]:
```suggestion
  "displayName": "Image",
```

**Review by nmadruga — CHANGES_REQUESTED** [review]
That's nice to have.

Just the test isn't the easiest. I think it would be good to add a simpler version of image displayed that we can easy change the size, alignment. It's not working on the go (which the provided test) on Section 2 Page 3

**Comment by CarlosNZ** [comment]
> Just the test isn't the easiest. I think it would be good to add a simpler version of image displayed that we can easy change the size, alignment. It's not working on the go (which the provided test) on Section 2 Page 3

The dynamic switching of the image should totally be working, but you might have to do a full reset (cache flush etc.) and database_init. I had to sometimes, but it always worked for me after that.

**Comment by nmadruga** [comment]
> The dynamic switching of the image should totally be working, but you might have to do a full reset (cache flush etc.) and database_init. I had to sometimes, but it always worked for me after that.

File: src/formElementPlugins/imageDisplay/pluginConfig.json (hunk 3 above)
Comment (CarlosNZ) [reply src/formElementPlugins/imageDisplay/pluginConfig.json:5]: Again -- do you really think something like "altText" needs to be dynamic?

**Review by nmadruga — APPROVED** [review]

**Comment by nmadruga** [comment]
Just made some comments for you and you will need to resolve the conflicts on Element-Type-Specs

File: documentation/internal/Element-Type-Specs.md (hunk 1 above)
Comment (CarlosNZ) [reply documentation/internal/Element-Type-Specs.md]: Good catch 😉 -- Fixed in latest commit (merged from below)

File: src/formElementPlugins/imageDisplay/pluginConfig.json (hunk 3 above)
Comment (nmadruga) [reply src/formElementPlugins/imageDisplay/pluginConfig.json:5]: Hmm maybe I didn't get this field correctly, I though it was listing fields that could be passed in to this plugin question

Comment (CarlosNZ) [reply src/formElementPlugins/imageDisplay/pluginConfig.json:5]: No, this is just the list of fields that the Wrapper will assume could be evaluator expressions, so will run evaluator on them. Fields not listed here will be passed through to plugin as is.

File: src/formElementPlugins/pluginProvider.tsx (hunk 2 above)
Comment (CarlosNZ) [reply src/formElementPlugins/pluginProvider.tsx:51]: I dunno, it kind of seems like a parallel name to "Text Info". Just "Image" as a name is a bit too ambiguous, don't you think? Could be image upload, etc.

**Pull request merged**